File: Quorum3Plugins/GamePlugin/JavaGame/src/plugins/quorum/Libraries/Game/Graphics/Renderable.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package plugins.quorum.Libraries.Game.Graphics;

import quorum.Libraries.Game.Graphics.Camera_;
import quorum.Libraries.Game.Graphics.Renderable_;

/**
 *
 * @author alleew
 */
public class Renderable 
{
    public java.lang.Object me_ = null;
    
    public Camera_ camera;
    
    public Shader shader;
    
    public Camera_ GetCamera()
    {
        return camera;
    }
    
    public void SetNative(Renderable_ other)
    {
        shader = ((quorum.Libraries.Game.Graphics.Renderable)other).plugin_.shader;
    }
    
}

File: Quorum3/Library/Standard/Plugins/javascript/Libraries/Game/Graphics/Renderable.js
function plugins_quorum_Libraries_Game_Graphics_Renderable_() 
{
    /*
     The publicly used properties of this class are:
        this.camera - A Quorum Camera object
        this.shader - A plugin Shader object
     */
    
    this.SetNative$quorum_Libraries_Game_Graphics_Renderable = function(renderable) 
    {
        this.shader = renderable.plugin_.shader;
    };
    
    this.GetCamera = function() 
    {
        return this.camera;
    };
}
